include/openthread/platform/radio.h

```diff
@@ -856,6 +876,8 @@ otError otPlatRadioSleep(otInstance *aInstance);
 /**
  * Transition the radio from Sleep to Receive (turn on the radio).
  *
+ * If the platform supports `OT_RADIO_CAPS_RX_ON_WHEN_IDLE`, this also updates the RxOnWhenIdle flag to TRUE.
```

Comment:
This is a fundamental change. Not sure if this is correct and what was intended by `OT_RADIO_CAPS_RX_ON_WHEN_IDLE` in 
- #9554.